webapp/components/comments/EditCommentForm/index.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <ds-form v-model="form" @submit="handleSubmit">
+    <template slot-scope="{ errors }">
+      <ds-card>
+        <!-- <no-ssr> -->
+        <hc-editor ref="editor" :users="users" :value="form.content" @input="updateEditorContent"/>
+        <!-- </no-ssr> -->
+        <ds-space/>
+        <ds-flex :gutter="{ base: 'small', md: 'small', sm: 'x-large', xs: 'x-large' }">
+          <ds-flex-item :width="{ base: '0%', md: '50%', sm: '0%', xs: '0%' }"/>
+          <ds-flex-item :width="{ base: '40%', md: '20%', sm: '30%', xs: '30%' }">
+            <ds-button
+              :disabled="disabled"
+              ghost
+              class="cancelBtn"
+              @click.prevent="closeEditWindow"
+            >{{ $t('actions.cancel') }}</ds-button>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '40%', md: '20%', sm: '40%', xs: '40%' }">
+            <ds-button
+              type="submit"
+              :loading="loading"
+              :disabled="disabled || errors"
+              primary
+            >{{ $t('post.comment.submit') }}</ds-button>
+          </ds-flex-item>
+        </ds-flex>
+      </ds-card>
+    </template>
+  </ds-form>
+</template>
+
+<script>
+import gql from 'graphql-tag'
+import HcEditor from '~/components/Editor'
+
+export default {
+  components: {
+    HcEditor
+  },
+  props: {
+    post: { type: Object, default: () => {} },
+    comments: { type: Array, default: () => [] },
+    comment: {
+      type: Object,
+      default() {
+        return {}
+      }
+    }
+  },
+  data() {
+    return {
+      disabled: false,
+      loading: false,
+      form: {
+        content: this.comment.contentExcerpt
+      },
+      users: []
+    }
+  },
+  methods: {
+    updateEditorContent(value) {
+      const content = value.replace(/<(?:.|\n)*?>/gm, '').trim()
+      if (content.length < 1) {
+        this.disabled = true
+      } else {
+        this.disabled = false
+      }
+      this.form.content = value
+    },
+    closeEditWindow() {
+      this.$emit('showEditCommentMenu', false)
+    },
+    handleSubmit() {
+      this.loading = true
+      this.disabled = true
+      this.$apollo
+        .mutate({
+          mutation: gql`
+            mutation($postId: ID, $content: String!, $id: ID!) {
+              UpdateComment(postId: $postId, content: $content, id: $id) {
+                id
+                content
+              }
+            }
+          `,
+          variables: {
+            postId: this.post.id,
+            content: this.form.content,
+            id: this.comment.id
+          }
+        })
+        .then(res => {
+          this.loading = false
+          this.$root.$emit('refetchPostComments')
+          this.$toast.success(this.$t('post.comment.updated'))
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-03T11:15:14Z" title="Monday, June 3rd 2019, 1:15:14 pm +02:00">Jun 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,121 @@
+<template>
+  <ds-form v-model="form" @submit="handleSubmit">
+    <template slot-scope="{ errors }">
+      <ds-card>
+        <!-- <no-ssr> -->
+        <hc-editor ref="editor" :users="users" :value="form.content" @input="updateEditorContent"/>
+        <!-- </no-ssr> -->
+        <ds-space/>
+        <ds-flex :gutter="{ base: 'small', md: 'small', sm: 'x-large', xs: 'x-large' }">
+          <ds-flex-item :width="{ base: '0%', md: '50%', sm: '0%', xs: '0%' }"/>
+          <ds-flex-item :width="{ base: '40%', md: '20%', sm: '30%', xs: '30%' }">
+            <ds-button
+              :disabled="disabled"
+              ghost
+              class="cancelBtn"
+              @click.prevent="closeEditWindow"
+            >{{ $t('actions.cancel') }}</ds-button>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '40%', md: '20%', sm: '40%', xs: '40%' }">
+            <ds-button
+              type="submit"
+              :loading="loading"
+              :disabled="disabled || errors"
+              primary
+            >{{ $t('post.comment.submit') }}</ds-button>
+          </ds-flex-item>
+        </ds-flex>
+      </ds-card>
+    </template>
+  </ds-form>
+</template>
+
+<script>
+import gql from 'graphql-tag'
+import HcEditor from '~/components/Editor'
+
+export default {
+  components: {
+    HcEditor
+  },
+  props: {
+    post: { type: Object, default: () => {} },
+    comments: { type: Array, default: () => [] },
+    comment: {
+      type: Object,
+      default() {
+        return {}
+      }
+    }
+  },
+  data() {
+    return {
+      disabled: false,
+      loading: false,
+      form: {
+        content: this.comment.contentExcerpt
+      },
+      users: []
+    }
+  },
+  methods: {
+    updateEditorContent(value) {
+      const content = value.replace(/<(?:.|\n)*?>/gm, '').trim()
+      if (content.length < 1) {
+        this.disabled = true
+      } else {
+        this.disabled = false
+      }
+      this.form.content = value
+    },
+    closeEditWindow() {
+      this.$emit('showEditCommentMenu', false)
+    },
+    handleSubmit() {
+      this.loading = true
+      this.disabled = true
+      this.$apollo
+        .mutate({
+          mutation: gql`
+            mutation($postId: ID, $content: String!, $id: ID!) {
+              UpdateComment(postId: $postId, content: $content, id: $id) {
+                id
+                content
+              }
+            }
+          `,
+          variables: {
+            postId: this.post.id,
+            content: this.form.content,
+            id: this.comment.id
+          }
+        })
+        .then(res => {
+          this.loading = false
+          this.$root.$emit('refetchPostComments')
+          this.$toast.success(this.$t('post.comment.submitted'))
```


hey @ALau2088, I was just having a look at this, and the functionality is looking good! One thing I noticed is that I think it might be nice to have a different message when the update was successful, could you add an entry for `post.comment.updated` in `en.json` and use it here?

Reply:
> [<img alt="ALau2088" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ALau2088) **Authored by [ALau2088](https://github.com/ALau2088)**
_<time datetime="2019-06-06T01:35:03Z" title="Thursday, June 6th 2019, 3:35:03 am +02:00">Jun 6, 2019</time>_

---


@mattwr18 . Do I pull down the latest by running _git pull upstream 552-update_comment_ ?

Reply:
> [<img alt="ALau2088" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ALau2088) **Authored by [ALau2088](https://github.com/ALau2088)**
_<time datetime="2019-06-06T01:43:44Z" title="Thursday, June 6th 2019, 3:43:44 am +02:00">Jun 6, 2019</time>_

---


@mattwr18 . The post.comment.updated has been added.